api/services/v1alpha1/gateway_types.go

```diff
@@ -40,31 +41,32 @@ type GatewayConfigSpec struct {
 	// +optional
 	OIDC *OIDCConfig `json:"oidc,omitempty"`
 
-	// Certificate management
+	// Certificate specifies configuration of the TLS certificate securing communication for the gateway.
 	// +optional
 	Certificate *infrav1.CertificateSpec `json:"certificate,omitempty"`
 
-	// Domain configuration for the GatewayConfig
-	// Example: apps.example.com
+	// Domain specifies the host name for intercepting incoming requests.
+	// Most likely, you will want to use a wildcard name, like *.example.com.
+	// If not set, the domain of the OpenShift Ingress is used.
+	// If you choose to generate a certificate, this is the domain used for the certificate request.
+	// Example: *.example.com, example.com, apps.example.com
 	// +optional
+	// +kubebuilder:validation:Pattern=`^(\*\.)?([a-z0-9]([-a-z0-9]*[a-z0-9])?\.)*[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
 	Domain string `json:"domain,omitempty"`
 
 	// Subdomain configuration for the GatewayConfig
+	// Example: my-gateway, custom-gateway
 	// +optional
-	// +kubebuilder:validation:MaxLength=63
 	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?)$`
 	Subdomain string `json:"subdomain,omitempty"`
 
-	// Cookie configuration for OAuth2 proxy (applies to both OIDC and OpenShift OAuth)
-	// +optional
-	Cookie *CookieConfig `json:"cookie,omitempty"`
+	// Cookie configuration (applies to both OIDC and OpenShift OAuth)
+	Cookie CookieConfig `json:"cookie"` // not pointer to make defaults clear
 
-	// AuthTimeout is the duration Envoy waits for auth proxy responses.
-	// Requests timeout with 403 if exceeded.
-	// Overrides GATEWAY_AUTH_TIMEOUT env var. Default: "5s"
+	// AuthProxyTimeout defines the timeout for external authorization service calls (e.g., "5s", "10s")
+	// This controls how long Envoy waits for a response from the authentication proxy before timing out 403 response.
 	// +optional
-	// +kubebuilder:validation:Pattern=`^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$`
-	AuthTimeout string `json:"authTimeout,omitempty"`
+	AuthProxyTimeout metav1.Duration `json:"authTimeout,omitempty"`
```

Comment:
The field rename from `authTimeout` to `authProxyTimeout` will cause silent data loss for existing configurations. We should mark `authTimeout` as deprecated and add a conversion logic in the controller.

Reply:
Originally, i wanna rename 
struct:
AuthTimeout => AuthProxyTimeout
API:
authTimeout => authProxyTimeout

but then i realized API change will be difficult. 
so i reverted authProxyTimeout back to old authTimeout